pyproject.toml

```diff
@@ -50,21 +50,24 @@ python = "^3.7"
 poetry-core = "1.4.0"
 poetry-plugin-export = "^1.2.0"
 "backports.cached-property" = { version = "^1.0.2", python = "<3.8" }
+build = "^0.9.0"
 cachecontrol = { version = "^0.12.9", extras = ["filecache"] }
 cleo = "^2.0.0"
 crashtest = "^0.4.1"
 dulwich = "^0.20.46"
 filelock = "^3.8.0"
 html5lib = "^1.0"
 importlib-metadata = { version = ">=4.4,<6", python = "<3.10" }
+installer = "^0.6.0"
 jsonschema = "^4.10.0"
 keyring = "^23.9.0"
 lockfile = "^0.12.2"
 # packaging uses calver, so version is unclamped
-packaging = ">=20.4"
+packaging = ">=20.4,<22.0"  # TODO remove upper bound when poetry-core with vendored packaging 22.0 is available
```

Comment:
why?

if it is necessary for poetry and poetry-core to agree on their packaging version then this ought to be an exact pin.  (I don't think this is necessary)

Reply:
Without it `test_prepare_directory_with_extensions` fails on macOS:

```
 >       assert wheel.name == f"extended-0.1-{env.supported_tags[0]}.whl"
E       AssertionError: assert 'extended-0.1...16_x86_64.whl' == 'extended-0.1..._0_x86_64.whl'
E         - extended-0.1-cp39-cp39-macosx_12_0_x86_64.whl
E         ?                                ^ ^
E         + extended-0.1-cp39-cp39-macosx_10_16_x86_64.whl
E         ?                                ^ ^^
```

Maybe, it's just an issue of the test. I'm not sure if changing the test to not expect the first element but just any element of `supported_tags` makes sense?

Reply:
Ah I see, poetry-core is building a platform-specific wheel using its packaging, and poetry tests are verifying the answer with a different packaging.

Meh, you're probably right just to do something pragmatic and not worry about it too much.